mathesar/views/api.py

```diff
@@ -331,19 +331,31 @@ def create(self, request, table_pk=None):
         table = get_table_or_404(table_pk)
         # We only support adding a single record through the API.
         assert isinstance((request.data), dict)
-        record = table.create_record_or_records(request.data)
-        serializer = RecordSerializer(record)
+        try:
+            record = table.create_record_or_records(request.data)
+            serializer = RecordSerializer(record)
+        except NotUniquePrimaryKey as e:
```

Comment:
We should be able to create records even if there's no primary key. Creation doesn't depend on a primary key.

Reply:
This probably needs to go on the `retrieve` method instead.